xds/internal/xdsclient/xdsresource/type_rds.go

```diff
@@ -144,6 +144,10 @@ type Route struct {
 	RetryConfig              *RetryConfig
 
 	RouteAction RouteAction
```

Comment:
`s/RouteAction/RouteActionType/g` in another PR?  @menghanl thoughts?  I'm confused that we have a RouteAction type that doesn't line up with Envoy's RouteAction message.

Reply:
Renaming SGTM. Also rename the field maybe: `ActionType RouteActionType`

The `Route` struct is the `RouteAction` message. It was OK since we didn't support other action types.
We just don't have a good way for oneof, or we make everything interfaces.

Reply:
Switched to ActionType RouteActionType.